monai/transforms/post/array.py

```diff
@@ -46,3 +47,68 @@ def __call__(self, img, to_onehot=None, num_classes=None):
             outputs.append(img[:, i : i + 1])
 
         return outputs
+
+
+class ConvertForMetrics(Transform):
+    """Execute after model forward to transform model output and labels for Ignite metrics.
+    It can complete below operations:
+        #. add `sigmoid` or `softmax` to y_pred
+        #. do `argmax` for y_pred
+        #. round y_pred value to 0.0 or 1.0
+        #. convert y_pred or y to One-Hot format
+
+    Args:
+        add_sigmoid (bool): whether to add sigmoid function to y_pred before transform.
+        add_softmax (bool): whether to add softmax function to y_pred before transform.
+        add_argmax (bool): whether to add argmax function to y_pred before transform.
+        to_onehot_y_pred (bool): whether to convert `y_pred` into the one-hot format. Defaults to False.
+        to_onehot_y (bool): whether to convert `y` into the one-hot format. Defaults to False.
+        n_classes (bool): the number of classes to convert to One-Hot format, if None, use `y_pred.shape[1]`
+        round_values (bool): whether round the value to 0 and 1, default is False.
+        logit_thresh (float): the threshold value to round value to 0.0 and 1.0, default is 0.5.
+
+    """
+
+    def __init__(
+        self,
+        add_sigmoid=False,
+        add_softmax=False,
+        add_argmax=False,
+        to_onehot_y_pred=False,
+        to_onehot_y=False,
+        n_classes=None,
+        round_values=False,
+        logit_thresh=0.5,
+    ):
+        self.add_sigmoid = add_sigmoid
+        self.add_softmax = add_softmax
+        self.add_argmax = add_argmax
+        self.to_onehot_y_pred = to_onehot_y_pred
+        self.to_onehot_y = to_onehot_y
+        self.n_classes = n_classes
+        self.round_values = round_values
+        self.logit_thresh = logit_thresh
+
+    def __call__(self, y_pred, y=None):
```

Comment:
I think this module does too many things, could decompose it into:
- `AsDiscrete`, for `argmax`, `to_one_hot`, `thresholding`, (round_values is not very accurate here)
- `NonLinear`, for `sigmoid`,`softmax`, other layers from `torch.nn`'s non linear functions collection https://pytorch.org/docs/stable/nn.html#non-linear-activations-other

also optionally we could have some basic input validation, e.g. sigmoid=True and softmax=True are not compatible options

Reply:
updated in the latest patch.
Thanks.